src/mjolnir/directededgebuilder.cc

```diff
@@ -163,6 +163,16 @@ DirectedEdgeBuilder::DirectedEdgeBuilder(const OSMWay& way,
       }
     }
   }
+  // Explicitely turn off access for constructions. It's done for backward compatibility
+  // of valhalla tiles and valhalla routing. In case we allow non-zero access then older
+  // versions of router will work with new tiles incorrectly. They would start to route
+  // on roads under construction because they're not awary about new 'Use::kConstruction'
+  // and use only access mode to check if an edge is routable or not.
+  if (way.use() == Use::kConstruction) {
+    forward_access = 0;
+    reverse_access = 0;
+    LOG_DEBUG("Turn off access mode for road under construction.");
```

Comment:
if we left this lua in:
https://github.com/valhalla/valhalla/pull/3455/files#r767592222

then we wouldnt need to do this here right? or does something further down in the updated lua turn access back on?

Reply:
yes, there is a bunch of logic in the lua function (`function filter_tags_generic(kv)`) that may override access tags. Moreover, some access tags may be changed in c++ code that goes after `parse_ways` stage.